pkg/engine/internal/imageverifier.go

```diff
@@ -422,20 +424,26 @@ func (iv *ImageVerifier) verifyAttestations(
 				}
 
 				attestationError = iv.verifyAttestation(cosignResp.Statements, attestation, imageInfo)
-				if attestationError != nil {
-					attestationError = fmt.Errorf("%s: %w", entryPath+subPath, attestationError)
-					return engineapi.RuleFail(iv.rule.Name, engineapi.ImageVerify, attestationError.Error()), ""
-				}
 
-				verifiedCount++
-				if verifiedCount >= requiredCount {
-					iv.logger.V(2).Info("image attestations verification succeeded", "verifiedCount", verifiedCount, "requiredCount", requiredCount)
-					break
+				if attestationError == nil {
+					verifiedCount++
+					if verifiedCount >= requiredCount {
+						iv.logger.V(2).Info("image attestations verification succeeded", "verifiedCount", verifiedCount, "requiredCount", requiredCount)
+						break
+					}
+				} else {
+					attestationError = fmt.Errorf("%s: %w", entryPath+subPath, attestationError)
+					errorList = append(errorList, attestationError)
 				}
 			}
 
+			err := multierr.Combine(errorList...)
+			errMsg := "attestations verification failed"
+			if err != nil {
+				errMsg = err.Error()
+			}
 			if verifiedCount < requiredCount {
```

Comment:
This means that we will only return errors if the required count is not met. From a UX perspective this makes sense to me. Is there a usecase where someone would want to know the failed attestations even if the required count has passed?

Reply:
> This means that we will only return errors if the required count is not met.
Yes, this is because we cannot return error when one entry fails. If required count is not met then we will send a combined error of all the attestation failure

>  someone would want to know the failed attestations even if the required count has passed?
Great point! I have added a log that will be logged when attestation verification fails for an entry.

Reply:
Please resolve comments if they are addressed.

Reply:
@vishal-chdhry - is this resolved?